src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -0,0 +1,248 @@
+import { Omnibox, IOmniboxSuggestion } from '../Omnibox/Omnibox';
+import { Component } from '../Base/Component';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { Assert } from '../../misc/Assert';
+import { Utils } from '../../utils/Utils';
+import { OmniboxEvents, IPopulateOmniboxSuggestionsEventArgs } from '../../events/OmniboxEvents';
+import { Initialization } from '../Base/Initialization';
+import { analyticsActionCauseList, IAnalyticsNoMeta } from '../Analytics/AnalyticsActionListMeta';
+import { $$ } from '../../utils/Dom';
+import { exportGlobally } from '../../GlobalExports';
+import 'styling/_FieldSuggestions';
+import * as _ from 'underscore';
+import { SuggestionsCache } from '../../misc/SuggestionsCache';
+import { QueryStateModel } from '../../ModelsModules';
+import { DomUtils } from '../../UtilsModules';
+import { IFacetValueSuggestionRow, FacetValueSuggestionsProvider, IFacetValueSuggestionsProvider } from './FacetValueSuggestionsProvider';
+
+export interface IFacetValueSuggestionsOptions {
+  numberOfSuggestions: number;
+  field?: IFieldOption;
+  useQuerySuggestions?: boolean;
+  useValueFromSearchbox?: boolean;
+  displayEstimateNumberOfResults?: boolean;
+  templateHelper?: (row: IFacetValueSuggestionRow, omnibox: Omnibox) => string;
+}
+
+/**
+ * The `FieldValueSuggestions` component provides query suggestions based on a particular field values.
+ *
+ * For example, if you use a `@category` field, this component will provide suggestions for categories that returns results for the given keywords.
+ *
+ * The query suggestions provided by this component appear in the [`Omnibox`]{@link Omnibox} component.
+ */
+export class FacetValueSuggestions extends Component {
+  static ID = 'FacetValueSuggestions';
+
+  static doExport = () => {
+    exportGlobally({
+      FacetValueSuggestions: FacetValueSuggestions
+    });
+  };
+
+  /**
+   * @componentOptions
+   */
+  static options: IFacetValueSuggestionsOptions = {
+    /**
+     * Specifies the facet field from which to provide suggestions.
+     *
+     * Specifying a value for this option is required for the `FieldValueSuggestions` component to work.
+     */
+    field: ComponentOptions.buildFieldOption({ required: true }),
+
+    /**
+     * Specifies the number of suggestions to render in the [`Omnibox`]{@link Omnibox}.
+     *
+     * Default value is `5`. Minimum value is `1`.
+     */
+    numberOfSuggestions: ComponentOptions.buildNumberOption({ defaultValue: 5, min: 1 }),
+
+    /**
+     * Specifies whether to use query suggestions as keywords to get facet values suggestions.
+     *
+     * Default value is `true`.
+     *
+     * **Note:**
```

Comment:
Anything we can do to make this simpler to configure, instead of relying on people reading the documentation (which they rarely do) ?

Could the component do check on "afterInitialization", and force the option if needed ?

Reply:
Personally not a fan of activating an option on another component since the default value is `true`.

Say I drop the following markup: `<div class="FacetValueSuggestions" data-field="@myfield"></div>`, it is not obvious that it will force `omnibox.enableQuerySuggestAddon` to `true`.